File: backend/models/deal.go
package models

import (
	"gorm.io/gorm"
)

type Deal struct {
	gorm.Model
	Title       string  `json:"title"`
	Description string  `json:"description"`
	Value       float64 `json:"value"`
	Status      string  `json:"status"`
	ContactID   uint    `json:"contact_id"`
	Contact     Contact `json:"contact" gorm:"foreignKey:ContactID"`
	UserID      uint    `json:"user_id"`
	User        User    `json:"user" gorm:"foreignKey:UserID"`
	Tasks       []Task  `json:"tasks" gorm:"foreignKey:DealID"`
}

File: frontend/app/src/types/Deal.ts
import { User } from './User';
import { Contact } from './Contact';
import { Task } from './Task';

export interface Deal {
  id: number;
  title: string;
  description: string;
  value: number;
  status: 'open' | 'won' | 'lost';
  stage: string; // Hinzugefügt um die Anforderungen von DealStats.tsx zu erfüllen
  expected_date: string;
  contact_id: number;
  user_id: number;
  created_at: string;
  updated_at: string;
  contact?: Contact;
  user?: User;
  tasks?: Task[];
}

export interface DealFormData {
  title: string;
  description: string;
  value: number;
  status: 'open' | 'won' | 'lost';
  expected_date: string;
  contact_id: number;
}
